### PR TITLE
[new] Navbar dropdown component

### LINKED ISSUE #3244: Update Navigation
See the [Figma mockup](https://www.figma.com/file/XL8jwSznvF6OFYTJkT9pYO/Barbara-x-Gitpod%3A-overhaul-website?node-id=251%3A907&t=EL3H82ZdYyNLHJGS-4)

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description

[**Preview**](https://deploy-preview-3262--gitpod-kumquat.netlify.app/)

Open Navbar Menu Items (Docs, Resources) in Both Web/Mobile mode.

## Issue

fixes #3244

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3262"></a>

### CODE REVIEW

File: src/lib/components/main-nav/docs-dropdown.svelte [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    return () => {
+      window.removeEventListener("click", handleClickOutside);
+    };
+  });
+</script>
+
+<style lang="postcss">
+  /* TODO: call this element to show it as selected */
+  .selected-menu {
```
Comment (Siddhant-K-code) [inline src/lib/components/main-nav/docs-dropdown.svelte:34]: @Cahllagerfeld Can you please help me out, how can we implement this part _(bottom border while selected)_?

image

File: src/lib/components/main-nav/docs-dropdown.svelte [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          aria-selected={false}
+          on:click={() => (shown = false)}
+          {href}
+        >
+          <p class="text-important font-bold mb-1">{text}</p>
+          <p>{description}</p>
+        </a>
+      {/each}
+    </div>
+    <div class="border-l border-divider py-micro">
```
Comment (Siddhant-K-code) [inline src/lib/components/main-nav/docs-dropdown.svelte:111]: @Cahllagerfeld Any Idea, How can we reduce this vertical divider's height?

image

**Comment by Siddhant-K-code** [comment]
Designs are ready to review except for a minor change (commented above) which requires input from @Cahllagerfeld.

// cc: @chrifro

File: src/lib/components/main-nav/docs-dropdown.svelte (hunk 2 above)
Comment (Cahllagerfeld) [reply src/lib/components/main-nav/docs-dropdown.svelte:111]: I think you may need a child-div with a margin and apply the border to this element

File: src/lib/components/main-nav/docs-dropdown.svelte (hunk 1 above)
Comment (Cahllagerfeld) [reply src/lib/components/main-nav/docs-dropdown.svelte:34]: I would try here to get this with the border of the `li-element` 🤔

**Review by chrifro — COMMENTED** [review]
Nice work, looks great✨

A few comments:

- [ ] mobile docs submenu: align the order to "docs", "Get started" and "Help center"
- [ ] mobile docs submenu: "get started" doesn't have a hover state. The default font color should be grey not white
- [x] on mobile: "docs" and "resources" are missing the hover style

- [ ] I did some minor adjustments in the [mockup ](https://www.figma.com/file/XL8jwSznvF6OFYTJkT9pYO/Barbara-x-Gitpod%3A-overhaul-website?node-id=329%3A1647&t=ypo6eLPazsUec6XE-1)to improve the mobile view
- add more spacing above "templates"
- change the font color for "templates"
- extended the color bar behind "templates" to fill the whole width

Feel free to merge once the comments have been addressed 🚀

**Pull request merged**